File: reverseWords/ReverseWords.java
import java.util.Scanner;
import java.io.FileReader;

public class ReverseWords {
    public static void main(String[] args) throws Exception {
        Scanner scanner = new Scanner(new FileReader(args[0]));
        while (scanner.hasNextLine()) {
            System.out.println(reverse(scanner.nextLine()));
        }
    }

    private static String reverse(String line) {
        String[] words = line.split(" ");
        StringBuilder reversed = new StringBuilder();
        for (int i = words.length - 1; i >= 0; i--) {
            reversed.append(words[i]).append(" ");
        }

        return reversed.toString().trim();
    }
}

File: reverseWords/reverseWords.js
fs = require("fs");
var lines = fs.readFileSync(process.argv[2], 'utf-8').split("\r\n");

lines.forEach(function(line) {
    var reversed = "";
    line.split(" ").reverse().forEach(function(word) {
        reversed += word + " ";
    });
    console.log(reversed.trim());
});